src/daemon/config.rs

```diff
@@ -655,19 +661,21 @@ impl Config {
             return Err(ConfigError::other("Port number must be >1024"));
         }
 
-        if !self.service_uri.ends_with('/') {
-            return Err(ConfigError::other("service URI must end with '/'"));
-        } else {
-            uri::Https::from_str(&self.service_uri)
-                .map_err(|_| ConfigError::Other(format!("Invalid service uri: {}", self.service_uri)))?;
+        if self.ip != ConfigDefaults::ip() && self.service_uri.is_none() {
+            return Err(ConfigError::other("Please specify a URI for 'service_uri' if you enable binding to IP addresses other than 127.0.0.1"));
```

Comment:
Maybe be nice to the user and generate service_uri if the IP is changed to save them specifying the IP twice by hand in the config file?

Reply:
changed - will now use the ip address if it is specified, to generate the service_uri if it's not overridden in the config.